src/main/java/controller/LadderGame.java

```diff
@@ -0,0 +1,66 @@
+package controller;
+
+import domain.*;
+import view.InputView;
+import view.OutputView;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class LadderGame {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public LadderGame(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+        final Players players = readWithRetry(this::readPlayers);
+        final Targets targets = readWithRetry(() -> readTargets(players));
+        final Height height = readWithRetry(this::readHeight);
+        final Ladder ladder = new Ladder(height, PlayerCount.from(players.getCount()));
+        final LadderResults ladderResults = createLadderResults(players, ladder, targets);
+        outputView.printResult(players, ladder, targets);
+        findLadderResult(players, ladderResults);
+    }
+
+    private Players readPlayers() {
+        final List<String> players = inputView.inputPlayers();
+        return Players.from(players);
+    }
+
+    private Targets readTargets(Players players) {
+        final List<String> targets = inputView.inputTarget();
+        return Targets.from(targets, PlayerCount.from(players.getCount()));
+    }
+
+    private Height readHeight() {
+        final int height = inputView.inputHeight();
+        return new Height(height);
+    }
+
+    private LadderResults createLadderResults(Players players, Ladder ladder, Targets targets) {
+        HashMap<Player, Target> results = new HashMap<>();
```

Comment:
```suggestion
        Map<Player, Target> results = new HashMap<>();
```
구현체로 생성보다 인터페이스로 생성해주세요
이유는 한번 생각해보면 좋을 것 같네요
이펙티브 자바 item 64를 참고해도 좋을것 같습니다